### ISSUE TITLE
Method N4JSProjectName#equals() throwing exception when hovering in package.json

### ISSUE THREAD

**Issue by mor-n4** [eclipse/n4js#1553]
Method `N4JSProjectName#equals()` throws an exception when hyper links are computed in `package.json` files.

To reproduce:

1. create a new N4JS project using the wizard,
2. open the `package.json` file,
3. hold down the command key (cmd) and hover the mouse over the dependency section inside the `package.json` editor.

Will throw:
```
java.lang.IllegalArgumentException: Cannot compare to type java.lang.String
	at org.eclipse.n4js.projectModel.names.N4JSProjectName.equals(N4JSProjectName.java:87)
	at com.google.common.base.Objects.equal(Objects.java:52)
	at org.eclipse.n4js.ui.editor.hyperlinking.packagejson.PackageJsonHyperlinkHelperExtension.getProjectDescriptionLocationForName(PackageJsonHyperlinkHelperExtension.java:260)
	at org.eclipse.n4js.ui.editor.hyperlinking.packagejson.PackageJsonHyperlinkHelperExtension.hyperlinkToProjectProperty(PackageJsonHyperlinkHelperExtension.java:224)
	at org.eclipse.n4js.ui.editor.hyperlinking.packagejson.PackageJsonHyperlinkHelperExtension.getUriRegionPair(PackageJsonHyperlinkHelperExtension.java:149)
	at org.eclipse.n4js.ui.editor.hyperlinking.packagejson.PackageJsonHyperlinkHelperExtension.getHyperlinks(PackageJsonHyperlinkHelperExtension.java:81)
	at org.eclipse.n4js.json.ui.editor.hyperlinking.JSONHyperlinkHelperProvider.createHyperlinksByOffset(JSONHyperlinkHelperProvider.java:28)
	at org.eclipse.xtext.ui.editor.hyperlinking.DefaultHyperlinkDetector$1.exec(DefaultHyperlinkDetector.java:51)
	at org.eclipse.xtext.ui.editor.hyperlinking.DefaultHyperlinkDetector$1.exec(DefaultHyperlinkDetector.java:1)
	at org.eclipse.xtext.util.concurrent.IReadAccess.lambda$tryReadOnly$0(IReadAccess.java:56)
	at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:91)
...
```

Obviously, the implementation of `N4JSProjectName#equals()` is violating the contract of `Object#equals()`. However, this was probably done to find suspicious places in the code where different kinds of project names are confused. So, probably rather `PackageJsonHyperlinkHelperExtension#getProjectDescriptionLocationForName()` should be adjusted.

**Comment by szarnekow** [comment]
> probably rather PackageJsonHyperlinkHelperExtension#getProjectDescriptionLocationForName() should be adjusted.

Yes, that's for sure.

**Comment by szarnekow** [comment]
Fix is part of #1576
